README.md

```diff
@@ -1,119 +1,120 @@
+### `core`
 
-The container can then be built and started with
+Contains all absolutely essential building that is necessary to run a connector such as `TransferProcessManager`,
+`ProvisionManager`, `DataFlowManager`, various model classes, the protocol engine and the policy piece. While it is
+possible to build a connector with just the code from the `core` module, it will have very limited capabilities to
+communicate and to interact with a data space.
 
-```shell
-./gradlew startDemo
-```
+### `extensions`
 
-which will launch a docker container based on the previously built image.
+This contains code that extends the connector's core functionality with technology- or cloud-provider-specific code. For
+example a transfer process store based on Azure CosmosDB, a secure vault based on Azure KeyVault, etc. This is where
+technology- and cloud-specific implementations should go.
 
-# Setup Azure resources
+If someone were to create a configuration service based on Postgres, then the implementation should go into
+the `extensions/database/configuration-postgres` module.
 
-*NB: This section is currently out-of-date and will be revised shortly*
+### `launchers`
 
-A working connector instance will use several resources on Azure, all of which can be easily deployed using a so-called
-"genesis script" located at `./scripts/genesis.sh`. Most Azure resources are grouped together in so-called "resource
-groups" which makes management quite easy. The Genesis Script will take care of provisioning the most essential
-resources and put them in a resource group:
+Launchers are essentially connector packages that are runnable. What modules get included in the build (and thus: what
+capabilities a connector has) is defined by the `build.gradle.kts` file inside the launcher subdirectory. That's also
+where a Java class containing a `main` method should go. We will call that class a "runtime" and in order for the
+connector to become operational the `runtime` needs to perform several important tasks (="bootstrapping"). For an
+example take a look at
+[this runtime](launchers/basic/src/main/java/org/eclipse/dataspaceconnector/runtime/ConnectorRuntime.java)
 
-- KeyVault
-- Blob Store Account
-- an AKS cluster
-- two app registrations / service principals
+### `data-protocols`
 
-App registrations are used to authenticate apps against Azure AD and are secured with certificates, so before
-provisioning any resources, the Genesis Script will generate a certificate (interactively) and upload it to the app
-registrations.
+Contains implementations for communication protocols a connector might use, such as IDS.
 
-The script requires Azure CLI (`az`) and `jq` (a JSON processor) and you need to be logged in to Azure CLI. Once that's
-done, simply `cd scripts` and invoke
+### `samples`
 
-```bash
-./genesis.sh <optional-prefix>
-```
+This is similar to the extensions module, but focuses more on showing specific use cases rather than technology
```

Comment:
feels to fuzzy to me. Maybe "Contains code that demonstrates how the connector can be used in various scenarios. For example, it shows..."